assets/large_language_models/rag/components/src/chat_flow_with_varients_mlindex.json

```diff
@@ -0,0 +1,413 @@
+                "text-embedding-ada-002",
+                "text-search-ada-doc-001",
+                "text-search-ada-query-001"
+              ],
+              "enabled_by": "connection",
+              "enabled_by_type": [
+                "OpenAIConnection"
+              ],
+              "name": "model"
+            },
+            "input": {
+              "type": [
+                "string"
+              ],
+              "name": "input"
+            }
+          },
+          "description": "Use Open AI's embedding model to create an embedding vector representing the input text.",
+          "module": "promptflow.tools.embedding",
+          "function": "embedding",
+          "is_builtin": true
+        },
+        {
+          "name": "Vector Search",
+          "type": "python",
+          "inputs": {
+            "path": {
+              "name": "path",
+              "type": [
+                "string"
+              ]
+            },
+            "query": {
+              "name": "query",
+              "type": [
+                "list",
+                "string"
+              ]
+            },
+            "top_k": {
+              "name": "top_k",
+              "type": [
+                "int"
+              ],
+              "default": "5"
+            }
+          },
+          "module": "promptflow_vectordb.tool.vector_index_lookup",
+          "class_name": "VectorIndexLookup",
+          "function": "search",
+          "is_builtin": true
+        },
+        {
+          "name": "generate_prompt_context",
+          "type": "python",
+          "inputs": {
+            "search_result": {
+              "name": "search_result",
+              "type": [
+                "object"
+              ]
+            }
+          },
+          "source": "generate_prompt_context.py",
+          "code": "from typing import List\nfrom promptflow import tool\nfrom promptflow_vectordb.core.contracts import SearchResultEntity\n\n@tool\ndef generate_prompt_context(search_result: List[dict]) -> str:\n    def format_doc(doc: dict):\n        return f\"Content: {doc['Content']}\\nSource: {doc['Source']}\"\n    \n    SOURCE_KEY = \"source\"\n    URL_KEY = \"url\"\n    \n    retrieved_docs = []\n    for item in search_result:\n\n        entity = SearchResultEntity.from_dict(item)\n        content  = entity.text or \"\"\n        \n        source = \"\"\n        if entity.metadata is not None:\n            if SOURCE_KEY in entity.metadata:\n                if URL_KEY in entity.metadata[SOURCE_KEY]:\n                    source = entity.metadata[SOURCE_KEY][URL_KEY] or \"\"\n        \n        retrieved_docs.append({\n            \"Content\": content,\n            \"Source\": source\n        })\n    doc_string = \"\\n\\n\".join([format_doc(doc) for doc in retrieved_docs])\n    return doc_string\n",
+          "function": "generate_prompt_context",
+          "is_builtin": false
+        },
+        {
+          "name": "Prompt_varients",
```

Comment:
nit, there is a typo, it should be "variants" and not "varients". This will need to be fixed in the non-chat json file as well as flow_creation.py

Reply:
fixed